core/src/main/java/com/linecorp/armeria/client/AbstractEventLoopState.java

```diff
@@ -29,6 +29,8 @@ static AbstractEventLoopState of(List<EventLoop> eventLoops, int maxNumEventLoop
                                      DefaultEventLoopScheduler scheduler) {
         if (maxNumEventLoops == 1) {
             return new OneEventLoopState(eventLoops, scheduler);
+        } else if (maxNumEventLoops <= 4) {
```

Comment:
```suggestion
        }
        if (maxNumEventLoops <= 4) {
```
Where did the number `4` come from?

Reply:
Can we also add JMH benchmark for the heap-based one to compare the numbers before and after this PR?